gc/base/standard/ParallelGlobalGC.cpp

```diff
@@ -506,8 +539,11 @@ MM_ParallelGlobalGC::mainThreadGarbageCollect(MM_EnvironmentBase *env, MM_Alloca
 	compactedThisCycle = _compactThisCycle;
 #endif /* OMR_GC_MODRON_COMPACTION */
 
-	/* If the delegate has isAllowUserHeapWalk set, fix the heap so that it can be walked */
-	if (_delegate.isAllowUserHeapWalk() || env->_cycleState->_gcCode.isRASDumpGC()) {
+	/* If the delegate has isAllowUserHeapWalk set, or should prepare for a snapshot,
+	 * fix the heap so that it can be walked
+	 */
+	MM_GCCode *gcCode = &env->_cycleState->_gcCode;
```

Comment:
you can actually use a snapshot of gcCode (rather than a pointer to it) - we don't really change gcCode in a middle of a GC - it's just set by whoever triggered GC.
So you can declare it as:

const MM_GCCode gcCode

not that there is another user of it at around of 571